### PR TITLE
Enable FindBugs in the plugin

### PR DESCRIPTION
**Pull request by oleg-nenashev**

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
@reviewbybees

File: src/main/java/hudson/tasks/Mailer.java [hunk 1]
```diff
@@ -174,7 +176,27 @@ public BuildStepMonitor getRequiredMonitorService() {
     }
 
     private static Pattern ADDRESS_PATTERN = Pattern.compile("\\s*([^<]*)<([^>]+)>\\s*");
+    
+    /**
+     * @deprecated Use {@link #stringToAddress(java.lang.String, java.lang.String)}.
+     */
+    @Deprecated
+    @SuppressFBWarnings(value = "NM_METHOD_NAMING_CONVENTION", justification = "It's deprecated and required for API compatibility")
```
Comment (jtnord) [inline src/main/java/hudson/tasks/Mailer.java]: add an @Restricted annotation as well?

File: src/main/java/hudson/tasks/Mailer.java [hunk 2]
```diff
@@ -509,20 +546,25 @@ public FormValidation doSendTestMail(
...
                 if (!useSMTPAuth) {
                     smtpAuthUserName = null;
                     smtpAuthPasswordSecret = null;
                 }
                 
                 MimeMessage msg = new MimeMessage(createSession(smtpServer, smtpPort, useSsl, smtpAuthUserName, smtpAuthPasswordSecret));
-                msg.setSubject(Messages.Mailer_TestMail_Subject(++testEmailCount), charset);
-                msg.setText(Messages.Mailer_TestMail_Content(testEmailCount, Jenkins.getInstance().getDisplayName()), charset);
-                msg.setFrom(StringToAddress(adminAddress, charset));
+                msg.setSubject(Messages.Mailer_TestMail_Subject(incrementTestEmailCount()), charset);
```
Comment (jtnord) [inline src/main/java/hudson/tasks/Mailer.java]: I wonder what findbugs complains about = as the new code isn't thread safe, just like the old code...

File: src/main/java/hudson/tasks/Mailer.java [hunk 3]
```diff
@@ -610,5 +652,6 @@ public UserProperty newInstance(StaplerRequest req, JSONObject formData) throws
     /**
      * Debug probe point to be activated by the scripting console.
      */
+    @SuppressFBWarnings(value = "MS_SHOULD_BE_FINAL", justification = "It's used for debugging")
```
Comment (jtnord) [inline src/main/java/hudson/tasks/Mailer.java]: the script console can access private fields...  Groovy has no concept of private!

File: src/main/java/hudson/tasks/Mailer.java (hunk 2 above)
Comment (oleg-nenashev) [inline src/main/java/hudson/tasks/Mailer.java]: No, it didn't.
I've thought about it but finally decided that an improper calculation does not break anything even if it happens. BTW we can migrate to AtomicInteger.

Comment (jtnord) [inline src/main/java/hudson/tasks/Mailer.java]: `ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD` is the original issue :-)

File: src/main/java/hudson/tasks/Mailer.java (hunk 3 above)
Comment (oleg-nenashev) [inline src/main/java/hudson/tasks/Mailer.java]: Yes, but we cannot change the access mode. Probably, somebody uses these flags in Java-based plugins

**Comment by oleg-nenashev** [comment]
@jtnord Done

File: src/main/java/hudson/tasks/Mailer.java (hunk 3 above)
Comment (jtnord) [inline src/main/java/hudson/tasks/Mailer.java]: if some java plugin is setting the debug mode of the mailer it deserves to be broken IMHO.
If there is a legitimate use requiring this to be public then the justification and comment are incorrect.
I'll leave the decision on this to the maintainer.

**Comment by jtnord** [comment]
:bee:  the maintainer ( whoever that is) may want to check [this comment ](#discussion-diff-34738571)

**Comment by oleg-nenashev** [comment]
I was wearing the maintainers hat for a while during bugfixes for my previous company, but I've never claimed it formally. Wiki indicates @tfennelly as a maintainer, but I'm not sure it's correct.

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Pull request merged**